packages/grafana-ui/src/utils/fieldDisplay.ts

```diff
@@ -0,0 +1,266 @@
+  limit?: number; // if showing all values limit
+  stats: string[]; // when !values, pick one value for the whole field
+
+  defaults: Partial<Field>; // Use these values unless otherwise stated
+  override: Partial<Field>; // Set these values regardless of the source
+
+  // Could these be data driven also?
+  thresholds: Threshold[];
+  mappings: ValueMapping[];
+}
+
+const VAR_SERIES_NAME = '__series_name';
+const VAR_FIELD_NAME = '__field_name';
+const VAR_STAT = '__stat';
+
+function getTitleTemplate(title: string | undefined, stats: string[], data?: SeriesData[]): string {
+  // If the title exists, use it as a template variable
+  if (title) {
+    return title;
+  }
+  if (!data || !data.length) {
+    return 'No Data';
+  }
+
+  let fieldCount = 0;
+  for (const field of data[0].fields) {
+    if (field.type === FieldType.number) {
+      fieldCount++;
+    }
+  }
+
+  const parts: string[] = [];
+  if (stats.length > 1) {
+    parts.push('$' + VAR_STAT);
+  }
+  if (data.length > 1) {
+    parts.push('$' + VAR_SERIES_NAME);
+  }
+  if (fieldCount > 1 || !parts.length) {
+    parts.push('$' + VAR_FIELD_NAME);
+  }
+  return parts.join(' ');
+}
+
+export interface FieldDisplay {
+  field: Field;
+  display: DisplayValue;
+  sparkline?: GraphSeriesValue[][];
+  prefix?: string;
+  suffix?: string;
+}
+
+export interface GetFieldDisplayValuesOptions {
+  data?: SeriesData[];
+  fieldOptions: FieldDisplayOptions;
+  replaceVariables: InterpolateFunction;
+  sparkline?: boolean; // Calculate the sparkline
+  theme: GrafanaTheme;
+}
+
+export const DEFAULT_FIELD_DISPLAY_VALUES_LIMIT = 25;
+
+export const getFieldDisplayValues = (options: GetFieldDisplayValuesOptions): FieldDisplay[] => {
+  const { data, replaceVariables, fieldOptions, sparkline } = options;
+  const { defaults, override } = fieldOptions;
+  const stats = fieldOptions.stats.length ? fieldOptions.stats : [StatID.last];
+
+  const values: FieldDisplay[] = [];
+
+  if (data) {
+    const limit = fieldOptions.limit ? fieldOptions.limit : DEFAULT_FIELD_DISPLAY_VALUES_LIMIT;
+    const title = getTitleTemplate(fieldOptions.title, stats, data);
+    const scopedVars: ScopedVars = {};
+
+    for (let s = 0; s < data.length; s++) {
+      const series = data[s];
+      scopedVars[VAR_SERIES_NAME] = { text: 'Series', value: series.name ? series.name : `Series[${s}]` };
+
+      let timeColumn = -1;
+      if (sparkline) {
+        for (let i = 0; i < series.fields.length; i++) {
+          if (series.fields[i].type === FieldType.time) {
+            timeColumn = i;
+            break;
+          }
+        }
+      }
+
+      for (let i = 0; i < series.fields.length; i++) {
+        const field = getFieldProperties(defaults, series.fields[i], override);
+
+        // Show all number fields
+        if (field.type !== FieldType.number) {
+          continue;
+        }
+
+        scopedVars[VAR_FIELD_NAME] = { text: 'Field', value: field.name ? field.name : `Field[${s}]` };
```

Comment:
@torkelo -- rather than putting `__field_name`, is it possible/reasonable to just put `__field` and then get the name (and other props!) via:  `${__field.name}`?